### PR TITLE
Refactor CSharp GSDK and tests to use injected dependencies 

### PR DESCRIPTION
**Pull request by ravarnamsft**
- Removed unnecessary async modifiers.
- Used injected dependencies in constructors.
- Removed and/or refactored unreliable tests (or ones which had external dependencies like the file system.

### CODE REVIEW

File: csharp/GSDK_CSharp_Standard/InternalSdk.cs [hunk 1]
```diff
@@ -83,42 +85,31 @@ public Task StartAsync(bool debugLogs = false)
...
+            Logger.Log($"Instance Id: {serverId}");
 
-            _webClient = HttpClientFactory.CreateInstance($"http://{gsmsBaseUrl}/v1/sessionHosts/{instanceId}");
+            _httpClient = _httpClientFactory.CreateInstance($"http://{heartbeatEndpoint}/v1/sessionHosts/{serverId}");
 
-            _heartbeatRunning = true;
             _signalHeartbeatEvent.Reset();
             TransitionToActiveEvent.Reset();
 
             _heartbeatTask = Task.Run(HeartbeatAsync);
```
Comment (yzhao174) [inline csharp/GSDK_CSharp_Standard/InternalSdk.cs]: >Task.Run(HeartbeatAsync); [](start = 29, length = 25)

Is this intended as fireAndForget?

File: csharp/GSDK_CSharp_Standard/GameserverSDK.cs [hunk 2]
```diff
@@ -61,8 +55,7 @@ public static class GameserverSDK
         /// </returns>
         public static bool ReadyForPlayers()
         {
-            Task.WhenAll(_internalSdk.StartAsync())
-                .Wait();
+            _internalSdk.Start();
```
Comment (yzhao174) [inline csharp/GSDK_CSharp_Standard/GameserverSDK.cs]: >Start [](start = 25, length = 5)

this class calls Start() in every method, and Start() internally skip if it is called more than once. Essentially Start() is an initialization method. Is it possible to merge it with InternalSdk constructor or make a factory method for it?

Comment (ravarnamsft) [reply csharp/GSDK_CSharp_Standard/GameserverSDK.cs]: Yeah, i tried adding a static constructor where I could call start. However, the problem is the argument to the Start() method, which is currently left to the caller (game server). We will need to make it a config value in app.config or something and then we can do this.
I spent sometime on making it static constructor and then decided that we can tackle that one separately. Also, like you pointed out, we don't really need the InternalSDK class. That's for another time as well.

File: csharp/GSDK_CSharp_Standard/InternalSdk.cs (hunk 1 above)
Comment (ravarnamsft) [reply csharp/GSDK_CSharp_Standard/InternalSdk.cs]: Yeah, that's my understanding, it just keeps running in the background. We should probably make it better by handling exceptions like we do on the VMAgent (RunUntilCancelled), but this refactoring seemed an improvement nonetheless.

**Pull request merged**